pkg/dns/config/config.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	types "k8s.io/kubernetes/pkg/api/unversioned"
+
+	"k8s.io/kubernetes/pkg/dns/fed"
+)
+
+// Config populated either from the configuration source (command
+// line flags or via the config map mechanism).
+type Config struct {
+	types.TypeMeta
```

Comment:
I suggested that on day we might generify the type-conversion stuff enough to reuse.  You should document what values you expect in these fields.